ev3dev/motor.py

```diff
@@ -67,6 +67,58 @@
     raise Exception("Unsupported platform '%s'" % platform)
 
 
+class SpeedInteger(int):
+    pass
+
+
+class SpeedRPS(SpeedInteger):
+    """
+    Speed in rotations-per-second
+    """
+
+    def __str__(self):
+        return ("%d rps" % self)
+
+    def get_speed_pct(self, motor):
+        return motor.rps_to_speed(self)
+
+
+class SpeedRPM(SpeedInteger):
+    """
+    Speed in rotations-per-minute
+    """
+
+    def __str__(self):
+        return ("%d rpm" % self)
+
+    def get_speed_pct(self, motor):
+        return motor.rpm_to_speed(self)
+
+
+class SpeedDPS(SpeedInteger):
+    """
+    Speed in degrees-per-second
+    """
+
+    def __str__(self):
+        return ("%d dps" % self)
+
+    def get_speed_pct(self, motor):
+        return motor.dps_to_speed(self)
```

Comment:
Doesn't it seem a bit odd that we convert degrees-per-second into percentage of max speed just to convert it back? I think this is one of those cases where we should choose the native unit and base everything on that -- degrees per second are the reasonable "scientific" units whereas the percentage isn't.

Reply:
A little...but the fact that we set the motor speed in terms of dps and EV3-G uses percentages means there is going to be some funkyness.  I can live with converting this one and then later having it converted back, since it means we treat all of these SpeedInteger classes consistently.